contracts/serverless-v2/Gateways.sol

```diff
@@ -379,59 +364,51 @@ contract Gateways is
     }
 
     function _addChains(
-        bytes memory _enclavePubKey,
         uint256[] memory _chainIds
     ) internal {
         if(_chainIds.length == 0)
             revert GatewaysEmptyRequestedChains();
 
         for (uint256 index = 0; index < _chainIds.length; index++) {
-            _addChain(
-                _enclavePubKey, 
-                _chainIds[index]
-            );
+            _addChain(_chainIds[index]);
         }
     }
 
     function _removeChains(
-        bytes memory _enclavePubKey,
         uint256[] memory _chainIds
     ) internal {
         if(_chainIds.length == 0)
             revert GatewaysEmptyRequestedChains();
 
         for (uint256 index = 0; index < _chainIds.length; index++) {
-            _removeChain(
-                _enclavePubKey, 
-                _chainIds[index]
-            );
+            _removeChain(_chainIds[index]);
         }
     }
 
     function _addChain(
-        bytes memory _enclavePubKey,
         uint256 _chainId
     ) internal {
         if(requestChains[_chainId].contractAddress == address(0))
             revert GatewaysUnsupportedChain();
 
-        address enclaveKey = _pubKeyToAddress(_enclavePubKey);
-        uint256[] memory chainIdList = gateways[enclaveKey].chainIds;
+        address operator = _msgSender();
+        _isValidGateway(operator);
+        uint256[] memory chainIdList = gateways[operator].chainIds;
         for (uint256 index = 0; index < chainIdList.length; index++) {
             if(chainIdList[index] == _chainId)
                 revert GatewaysChainAlreadyExists(_chainId);
         }
-        gateways[enclaveKey].chainIds.push(_chainId);
+        gateways[operator].chainIds.push(_chainId);
 
-        emit ChainAdded(enclaveKey, _chainId);
+        emit ChainAdded(operator, _chainId);
     }
 
     function _removeChain(
-        bytes memory _enclavePubKey,
         uint256 _chainId
     ) internal {
-        address enclaveKey = _pubKeyToAddress(_enclavePubKey);
-        uint256[] memory chainIdList = gateways[enclaveKey].chainIds;
+        address operator = _msgSender();
+        _isValidGateway(operator);
```

Comment:
Can we do this check only once in _removeChains method. Otherwise it will be repeated for each chain in arg.